internal/server/data/identity.go

```diff
@@ -112,51 +131,329 @@ func AssignIdentityToGroups(tx GormTxn, user *models.Identity, provider *models.
 	return nil
 }
 
-func CreateIdentity(db GormTxn, identity *models.Identity) error {
-	return add(db, identity)
+func CreateIdentity(tx WriteTxn, identity *models.Identity) error {
+	if identity.VerificationToken == "" {
+		identity.VerificationToken = generate.MathRandom(10, generate.CharsetAlphaNumeric)
+	}
+	return insert(tx, (*identitiesTable)(identity))
 }
 
-func GetIdentity(db GormTxn, selectors ...SelectorFunc) (*models.Identity, error) {
-	return get[models.Identity](db, selectors...)
+type GetIdentityOptions struct {
+	ByID             uid.ID
+	ByName           string
+	PreloadGroups    bool
+	PreloadProviders bool
```

Comment:
What do you think about calling these `LoadGroups` and `LoadProviders` ?  `Preload` I think is a Gorm term that I was not really familiar with. In the future we might want to implement these using joins instead of separately selecting everything. We might only need to join across to the id and name (to avoid loading the full model of providers and groups). Calling it something other than `Preload` might help indicate that it's not necessarily the same as what gorm was doing.

If it's a hassle to change I think it's fine this way, we can change it later.

Reply:
Made this change, not a hassle at all